rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,1076 @@
+# 0000: Extend the PE subfield of the File fieldset
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.pe.authentihash | keyword | Authentihash of the PE file. |
+| file.pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| file.pe.compiler_product_versions | keyword | Version of the compiler. |
+| file.pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.pe.entry_point | long | Entry point of the PE file. |
+| file.pe.exports | keyword | List of symbols exported by PE |
+| file.pe.debug | flattened | Debug information, if present |
+| file.pe.import_list | flattened | List of all imported functions |
```

Comment:
I'd really like to present a common interface to imports across PE, MachO, and ELF. A project called LIEF does just this with respect to symbols and a base [abstract symbol](https://lief.quarkslab.com/doc/latest/api/python/abstract.html#symbol) definition. Basically, I propose a superset of what we have today from each of these three types from VT. Tools will populate the objects with information available and `file.*.imported_symbols` would be a `nested` list of objects with: `name`, `type`, `library_name` at a minimum. These 3 fields would capture everything VT provides today, but a lot of other frameworks (including LIEF) provide additional info, such as library offset, etc. This could offer a good future expansion of the fieldset. 

See additional discussion here: https://github.com/elastic/beats/pull/21815#discussion_r515411414

Reply:
Was thinking about this more @dcode -- might want to keep this as `type: flattened` field, mainly to avoid a glut of potential sub documents if this field was `nested`. I can imagine a binary or dynamic library that imports more than 1000 symbols. Maybe we could just figure out a way of documenting, "only use these sub fields in this flattened field"?

Also maybe rename to `imports`?